### PR TITLE
Add gallery example "Decorated lines" (`style="~"`)

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to add a gallery example showing how to generated "Decorated lines".

**Upstream GMT documentation**: https://docs.generic-mapping-tools.org/dev/plot.html#s

**Related to**: 
- Gallery example [Line fronts](https://www.pygmt.org/dev/gallery/lines/linefronts.html)  (`style="f"`)
- PR #2563

**Preview**:  https://pygmt-dev--2564.org.readthedocs.build/en/2564/gallery/lines/decorated_lines.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]
Looks good.

File: examples/gallery/lines/decorated_lines.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# Generate a two-point line for plotting
+x = np.array([1, 4])
+y = np.array([24, 24])
+
+fig = pygmt.Figure()
+fig.basemap(
+    region=[0, 10, 0, 24],
+    projection="X15c",
+    frame="+tDecorated Lines",
+)
```
Comment (seisman) [inline examples/gallery/lines/decorated_lines.py:32]:
```suggestion
fig.basemap(region=[0, 10, 0, 24], projection="X15c", frame="+tDecorated Lines")
```

**Comment by yvonnefroehlich** [comment]
Based on the upstream GMT documentation at https://docs.generic-mapping-tools.org/dev/plot.html#s it should be possible to rotate the symbol by appending **+a**_angle_:
> +aangle
> Force symbols at a fixed angle, +an for line-normal, or +ap for line-parallel [Default].

However, so far this does not work for me (for quoted lines it works to rotate the label by appending **+a**_angle_).

_Code example_:
```python
import pygmt
import numpy as np


size = 5
x = np.array([-size, size])
y = np.array([3, 3])

fig = pygmt.Figure()

fig.basemap(
    region=[-size, size, -size, size],
    projection="X" + str(size*2) + "c",
    frame=True,
)

for i_line in [
        "~d2c:+ss1c",
        "~d2c:+ss1c+a20",
        "qd1c:+ltext",
        "qd1c:+ltext+a20",
        ]:
    fig.plot(
        x=x,
        y=y,
        style=i_line,
        pen="1.25p,black",
    )
    
    fig.text(
        x=0,
        y=y[0]+0.7,
        text=i_line,
        font="10p",
    )
    
    y = y - 2
    
fig.show()
# fig.savefig(fname="decorated_lines_rotated.png")
```

_Output figure_:
decorated_lines_rotated

**Comment by seisman** [comment]
> Based on the upstream GMT documentation at https://docs.generic-mapping-tools.org/dev/plot.html#s it should be possible to rotate the symbol by appending **+a**_angle_:
> 
> > +aangle
> > Force symbols at a fixed angle, +an for line-normal, or +ap for line-parallel [Default].
> 
> However, so far this does not work for me (for quoted lines it works to rotate the label by appending **+a**_angle_).

Likely an upstream bug. Please report to the upstream GMT repository.

**Comment by yvonnefroehlich** [comment]
> > Based on the upstream GMT documentation at https://docs.generic-mapping-tools.org/dev/plot.html#s it should be possible to rotate the symbol by appending **+a**_angle_:
> > > +aangle
> > > Force symbols at a fixed angle, +an for line-normal, or +ap for line-parallel [Default].
> > 
> > 
> > However, so far this does not work for me (for quoted lines it works to rotate the label by appending **+a**_angle_).
> 
> Likely an upstream bug or a missing feature. Please report to the upstream GMT repository.

Please see issue https://github.com/GenericMappingTools/gmt/issues/7591.

File: examples/gallery/lines/decorated_lines.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+"""
+Decorated lines
+---------------
+To draw a so-called *decorated line*, i.e., symbols along a line
+or curve, use the ``style`` parameter of the
+:meth:`pygmt.Figure.plot` method with the argument ``"~"`` and the
+desired modifiers. A colon (``":"``) is used to separate the
+algorithm settings from the symbol information.
+ This example shows how to adjust the symbols.
+Beside the build-in symbols also custom symbols can be used.
```
Comment (michaelgrund) [inline examples/gallery/lines/decorated_lines.py:10]:
```suggestion
Beside the built-in symbols also custom symbols can be used.
```

File: examples/gallery/lines/decorated_lines.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+For modifying the base line via the ``pen`` parameter, see the
```
Comment (michaelgrund) [inline examples/gallery/lines/decorated_lines.py:11]:
```suggestion
For modifying the baseline via the ``pen`` parameter, see the
```

**Review by michaelgrund — APPROVED** [review]

File: examples/gallery/lines/decorated_lines.py (hunk 3 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/decorated_lines.py]: All occurencies are updated in commit https://github.com/GenericMappingTools/pygmt/pull/2564/commits/d0d7c59463bf72d8f5736abfaadca701cec68825.

**Pull request merged**